extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -368,40 +371,87 @@ export abstract class AzureAuth implements vscode.Disposable {
-			const tokenClaims = this.getTokenClaims(tokenResponse.data.access_token);
-
-			const accessToken: AccessToken = {
-				token: tokenResponse.data.access_token,
-				key: tokenClaims.email || tokenClaims.unique_name || tokenClaims.name,
-			};
-
-			this.memdb.set(this.createMemdbString(accessToken.key, tenant, resourceId), tokenResponse.data.expires_on);
+			let refreshResponse: TokenRefreshResponse;
 
-			const refreshToken: RefreshToken = {
-				token: tokenResponse.data.refresh_token,
-				key: accessToken.key
-			};
-
-			return { accessToken, refreshToken, tokenClaims };
+			try {
+				const tokenUrl = `${this.loginEndpointUrl}${tenant}/oauth2/token`;
+				const tokenResponse = await this.makePostRequest(tokenUrl, postData);
+				const tokenClaims = this.getTokenClaims(tokenResponse.data.access_token);
+
+				const accessToken: AccessToken = {
+					token: tokenResponse.data.access_token,
+					key: tokenClaims.email || tokenClaims.unique_name || tokenClaims.name,
+				};
+
+				const refreshToken: RefreshToken = {
+					token: tokenResponse.data.refresh_token,
+					key: accessToken.key
+				};
+				const expiresOn = tokenResponse.data.expires_on;
+
+				refreshResponse = { accessToken, refreshToken, tokenClaims, expiresOn };
+			} catch (ex) {
+				if (ex?.response?.data?.error === 'interaction_required') {
+					const shouldOpenLink = await this.openConsentDialog(tenant, resourceId);
+					if (shouldOpenLink === true) {
+						const { tokenRefreshResponse, authCompleteDeferred } = await this.promptForConsent(resourceEndpoint, tenant);
+						refreshResponse = tokenRefreshResponse;
+						authCompleteDeferred.resolve();
+					} else {
+						vscode.window.showInformationMessage(localize('azure.noConsentToReauth', "The authentication failed since ADS was unable to open re-authentication page."));
+					}
+				} else {
+					return undefined;
+				}
+			}
 
+			this.memdb.set(this.createMemdbString(refreshResponse.accessToken.key, tenant, resourceId), refreshResponse.expiresOn);
+			return refreshResponse;
 		} catch (err) {
 			const msg = localize('azure.noToken', "Retrieving the Azure token failed. Please sign in again.");
 			vscode.window.showErrorMessage(msg);
 			throw new Error(err);
 		}
 	}
 
+	private async openConsentDialog(tenantId: string, resourceId: string): Promise<boolean> {
+		return new Promise((resolve) => {
+			// 20 second timeout just in case something happens.
+			const timer = setTimeout(() => {
+				resolve(false);
+			}, 20 * 1000);
+
+			const dialogTitle = localize('azurecore.consentDialog.title', "Authentication Consent Confirmation");
+			const dialogBody = localize('azurecore.consentDialog.body', "Your tenant {0} requires you to re-authenticate again to access {1} resources. Press OK to start the authentication process.", tenantId, resourceId);
+			const dialog = azdata.window.createModelViewDialog(dialogTitle, 'azureCore.consentDialog');
```

Comment:
ModelViewDialog seems pretty overkill for this - why not just use showInformationMessage? 

https://github.com/microsoft/azuredatastudio/blob/master/src/vs/vscode.d.ts#L7351

Reply:
Is there a way to make the user have to press okay or something?

Reply:
Yes there is.

Reply:
How are you opening the web page? You could also use the opener service which would prompt for you. Granted it's not as descriptive/friendly as this but something to consider.

Reply:
Being as this is from an extension openExternalUrl should go through that IIRC